source/extensions/filters/network/common/redis/fault_impl.h

```diff
@@ -0,0 +1,79 @@
+#pragma once
+
+#include <map>
+#include <string>
+#include <vector>
+
+#include "envoy/api/api.h"
+#include "envoy/extensions/filters/network/redis_proxy/v3/redis_proxy.pb.h"
+#include "envoy/upstream/upstream.h"
+
+#include "common/protobuf/utility.h"
+
+#include "extensions/filters/network/common/redis/fault.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace Common {
+namespace Redis {
+
+class Fault {
+public:
+  Fault(envoy::extensions::filters::network::redis_proxy::v3::RedisProxy_RedisFault base_fault);
+
+private:
+  static std::vector<std::string> getCommands(
+      envoy::extensions::filters::network::redis_proxy::v3::RedisProxy_RedisFault base_fault);
+
+public:
+  FaultType fault_type_;
```

Comment:
just noticed these fields are public, this feels like an antipattern

Reply:
I'll add getters